pkg/execution/executor/executor.go

```diff
@@ -2556,11 +2561,7 @@ func (e *executor) handleGeneratorWaitForEvent(ctx context.Context, i *runInstan
 			"event": evt.Map(),
 		})
 		if err != nil {
-			logger.StdlibLogger(ctx).Warn(
-				"error interpolating waitForEvent expression",
-				"error", err,
-				"expression", *opts.If,
-			)
+			return fmt.Errorf("error interpolating wait for event expression: %w", err)
```

Comment:
This is where we currently end with a stuck function run. Now, we simply return the error and handle it a few layers above.